modules/elements-color-picker/assets/js/editor/commands/show-swatches.js

```diff
@@ -55,7 +55,20 @@ export class ShowSwatches extends CommandBase {
 			if ( isImage ) {
 				this.extractColorsFromImage( e.target );
 			} else {
+				// Colors from the parent container.
 				this.extractColorsFromSettings();
+
+				// Colors from repeaters.
+				if ( 0 !== Object.keys( this.container.repeaters ).length ) {
+					// Iterate over repeaters.
+					Object.values( this.container.repeaters ).forEach( ( repeater ) => {
+						// Iterate over each repeater items.
+						repeater.children.forEach( ( child ) => {
+							this.extractColorsFromSettings( child );
+						} );
+					} );
+				}
```

Comment:
Extract into method e.g: `this.extractColorsFromRepeater()`

Reply:
Yup, good idea :)